contribulyze/contribulyze.py

```diff
@@ -0,0 +1,663 @@
+    # parentheses and trailing newline are left on.  In theory, this
+    # supports concatenation of consecutive asides.  In practice, the
+    # parser only detects the first one anyway, because additional
+    # ones are very uncommon and furthermore by that point one should
+    # probably be looking at the full log message.
+    self.addendum = ''
+  def add_contributor(self, contributor):
+    self.contributors.add(contributor)
+  def add_endum(self, addendum):
+    self.addendum += addendum
+  def __str__(self):
+    s = 'FIELD: %s (%d contributors)\n' % (self.name, len(self.contributors))
+    for contributor in self.contributors:
+      s += str(contributor) + '\n'
+    s += self.addendum
+    return s
+
+
+class LogMessage(object):
+  # Maps sha strings onto LogMessage instances,
+  # holding all the LogMessage instances ever created.
+  all_logs = { }
+  author = None
+  date = None
+  message = ''
+  latest = None
+  def __init__(self, sha):
+    """Instantiate a log message.  All arguments are strings, including commit."""
+    self.sha = sha
+    # Map field names (e.g., "Patch", "Review") onto Field objects.
+    self.fields = { }
+    if sha in LogMessage.all_logs:
+      complain("SHA '%s' seen more than once" % sha, True)
+    LogMessage.all_logs[sha] = self
+  def add_field(self, field):
+    self.fields[field.name] = field
+  def accum(self, line):
+    """Accumulate one more line of raw message."""
+    self.message += line
+
+  def __cmp__(self, other):
+    """Compare two log messages by date, for sort().
+    Return -1, 0 or 1 depending on whether a > b, a == b, or a < b.
+    Note that this is reversed from normal sorting behavior, but it's
+    what we want for reverse chronological ordering of revisions."""
+    a = self.date
+    b = other.date
+    if a > b: return -1
+    if a < b: return 1
+    else:     return 0
+
+  def sort_key(self):
+    return self.date
+
+  def __str__(self):
+    s = '=' * 15
+    header = ' COMMIT: %s | %s \n %s' % (self.sha, self.author, self.message)
+    s += header
+    s += '=' * 15
+    s += '\n'
+    for field_name in self.fields.keys():
+      s += str(self.fields[field_name]) + '\n'
+    s += '-' * 15
+    s += '-' * len(header)
+    s += '-' * 15
+    s += '\n'
+    return s
+
+def process_aliases(aliases_input):
+  line = aliases_input.readline()
+  while line:
+    aliases = line.split(',')
+    c = Contributor.get(aliases.pop(0).strip(), None)
+    for alias in aliases:
+      c.add_aliases(alias.strip())
+    line = aliases_input.readline()
+
+def process_committers():
+    committers_url = 'https://whimsy.apache.org/public/public_ldap_projects.jsonn'
+    names_url = 'https://whimsy.apache.org/public/icla-info.json'
+    committers = json.loads(requests.get(committers_url).text)['projects']['cassandra']['members']
+    names_json = json.loads(requests.get(names_url).text)['committers']
+    for committer in committers:
+        name = names_json[committer]
```

Comment:
If the `committer` alias isn't in the names list, it should keep that as the name to be looked up.
```suggestion
        name = names_json.get(committer, committer)
```